tests: i2s: i2s_speed build failure on frdm_mcxn947/mcxn947/cpu0 platform

**Describe the bug**

i2s_speed build failure on frdm_mcxn947/mcxn947/cpu0 platform

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - frdm_mcxn947/mcxn947/cpu0
 - What have you tried to diagnose or workaround this issue?
 - gi bisect
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 
```
commit e1f578325dbe0fe67bb5db4ba691fff3fe28f237
Author: Declan Snyder <declan.snyder@nxp.com>
Date:   Thu Jan 16 17:03:13 2025 -0600

    dts: i2s_mcux_sai: Clarify IOMUXC GPR binding
    
    Improve the documentation of the IOMUXC GPR binding so that it is
    clear what the cells in the specifier space of "pinmux" space mean. And
    change the names of the cells to be more appropriate. "offset" instead
    of "pin" and "mask" instead of "function" describes more precisely what
    the cells in the specifier mean.
    
    Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

 drivers/i2s/i2s_mcux_sai.c            |  4 ++--
 dts/bindings/pinctrl/nxp,imx-gpr.yaml | 13 ++++++++++---
 2 files changed, 12 insertions(+), 5 deletions(-)
```


**To Reproduce**

```
scripts\twister -p frdm_mcxn947/mcxn947/cpu0 -T tests/drivers/i2s/i2s_speed 
```

**Expected behavior**

build pass

**Impact**

I2S function for this platform

**Logs and console output**

```
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_mcxn947_mcxn947_cpu0/zephyr/tests/drivers/i2s/i2s_speed/drivers.i2s.speed/zephyr/include/generated/zephyr/devicetree_generated.h:13089:34: error: 'DT_N_S_soc_S_peripheral_50000000_S_sai_106000_P_pinmuxes_IDX_0_VAL_mask' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_peripheral_50000000_S_sai_106000_P_pinmuxes_IDX_0_VAL_pin'?
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 13089 | #define DT_N_INST_0_nxp_mcux_i2s DT_N_S_soc_S_peripheral_50000000_S_sai_106000
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 72 | #define __DEBRACKET(...) __VA_ARGS__
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                          ^~~~~~~~~~~
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~~~
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:4956:9: note: in expansion of macro 'COND_CODE_1'
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 4956 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:1565:9: note: in expansion of macro 'DT_CAT7'
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1565 |         DT_CAT7(node_id, _P_, pha, _IDX_, idx, _VAL_, cell)
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1156:34: note: in expansion of macro 'DT_PHA_BY_IDX'
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1156 |                 .mclk_pin_mask = DT_PHA_BY_IDX(DT_DRV_INST(i2s_id), pinmuxes, 0, mask),            \
2025-01-26 16:03:51,272 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                  ^~~~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_0_nxp_mcux_i2s'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                    ^
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                          ^~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:349:31: note: in expansion of macro 'UTIL_CAT'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 349 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                               ^~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:3879:27: note: in expansion of macro 'DT_INST'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 3879 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                           ^~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1156:48: note: in expansion of macro 'DT_DRV_INST'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1156 |                 .mclk_pin_mask = DT_PHA_BY_IDX(DT_DRV_INST(i2s_id), pinmuxes, 0, mask),            \
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                                ^~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_mcxn947_mcxn947_cpu0/zephyr/tests/drivers/i2s/i2s_speed/drivers.i2s.speed/zephyr/include/generated/zephyr/devicetree_generated.h:27497:47: note: in expansion of macro 'I2S_MCUX_INIT'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 27497 | #define DT_FOREACH_OKAY_INST_nxp_mcux_i2s(fn) fn(0) fn(1)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                               ^~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_mcux_i2s'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                    ^
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1222:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1222 | DT_INST_FOREACH_STATUS_OKAY(I2S_MCUX_INIT)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_mcxn947_mcxn947_cpu0/zephyr/tests/drivers/i2s/i2s_speed/drivers.i2s.speed/zephyr/include/generated/zephyr/devicetree_generated.h:13089:34: error: 'DT_N_S_soc_S_peripheral_50000000_S_sai_106000_P_pinmuxes_IDX_0_VAL_offset' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_peripheral_50000000_S_sai_106000_P_pinmuxes_IDX_0_VAL_pin'?
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 13089 | #define DT_N_INST_0_nxp_mcux_i2s DT_N_S_soc_S_peripheral_50000000_S_sai_106000
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 72 | #define __DEBRACKET(...) __VA_ARGS__
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                          ^~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:4956:9: note: in expansion of macro 'COND_CODE_1'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 4956 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:1565:9: note: in expansion of macro 'DT_CAT7'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1565 |         DT_CAT7(node_id, _P_, pha, _IDX_, idx, _VAL_, cell)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1157:36: note: in expansion of macro 'DT_PHA_BY_IDX'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1157 |                 .mclk_pin_offset = DT_PHA_BY_IDX(DT_DRV_INST(i2s_id), pinmuxes, 0, offset),        \
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                    ^~~~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_0_nxp_mcux_i2s'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                    ^
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                          ^~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:349:31: note: in expansion of macro 'UTIL_CAT'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 349 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                               ^~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:3879:27: note: in expansion of macro 'DT_INST'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 3879 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                           ^~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1157:50: note: in expansion of macro 'DT_DRV_INST'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1157 |                 .mclk_pin_offset = DT_PHA_BY_IDX(DT_DRV_INST(i2s_id), pinmuxes, 0, offset),        \
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                                  ^~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_mcxn947_mcxn947_cpu0/zephyr/tests/drivers/i2s/i2s_speed/drivers.i2s.speed/zephyr/include/generated/zephyr/devicetree_generated.h:27497:47: note: in expansion of macro 'I2S_MCUX_INIT'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 27497 | #define DT_FOREACH_OKAY_INST_nxp_mcux_i2s(fn) fn(0) fn(1)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                               ^~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_mcux_i2s'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                    ^
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1222:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1222 | DT_INST_FOREACH_STATUS_OKAY(I2S_MCUX_INIT)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_mcxn947_mcxn947_cpu0/zephyr/tests/drivers/i2s/i2s_speed/drivers.i2s.speed/zephyr/include/generated/zephyr/devicetree_generated.h:13347:34: error: 'DT_N_S_soc_S_peripheral_50000000_S_sai_107000_P_pinmuxes_IDX_0_VAL_mask' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_peripheral_50000000_S_sai_107000_P_pinmuxes_IDX_0_VAL_pin'?
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 13347 | #define DT_N_INST_1_nxp_mcux_i2s DT_N_S_soc_S_peripheral_50000000_S_sai_107000
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 72 | #define __DEBRACKET(...) __VA_ARGS__
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                          ^~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~
2025-01-26 16:03:51,273 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:4956:9: note: in expansion of macro 'COND_CODE_1'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 4956 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:1565:9: note: in expansion of macro 'DT_CAT7'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1565 |         DT_CAT7(node_id, _P_, pha, _IDX_, idx, _VAL_, cell)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1156:34: note: in expansion of macro 'DT_PHA_BY_IDX'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1156 |                 .mclk_pin_mask = DT_PHA_BY_IDX(DT_DRV_INST(i2s_id), pinmuxes, 0, mask),            \
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                  ^~~~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_1_nxp_mcux_i2s'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                    ^
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                          ^~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:349:31: note: in expansion of macro 'UTIL_CAT'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 349 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                               ^~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:3879:27: note: in expansion of macro 'DT_INST'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 3879 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                           ^~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1156:48: note: in expansion of macro 'DT_DRV_INST'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1156 |                 .mclk_pin_mask = DT_PHA_BY_IDX(DT_DRV_INST(i2s_id), pinmuxes, 0, mask),            \
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                                ^~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_mcxn947_mcxn947_cpu0/zephyr/tests/drivers/i2s/i2s_speed/drivers.i2s.speed/zephyr/include/generated/zephyr/devicetree_generated.h:27497:53: note: in expansion of macro 'I2S_MCUX_INIT'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 27497 | #define DT_FOREACH_OKAY_INST_nxp_mcux_i2s(fn) fn(0) fn(1)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                                     ^~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_mcux_i2s'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                    ^
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1222:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1222 | DT_INST_FOREACH_STATUS_OKAY(I2S_MCUX_INIT)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_mcxn947_mcxn947_cpu0/zephyr/tests/drivers/i2s/i2s_speed/drivers.i2s.speed/zephyr/include/generated/zephyr/devicetree_generated.h:13347:34: error: 'DT_N_S_soc_S_peripheral_50000000_S_sai_107000_P_pinmuxes_IDX_0_VAL_offset' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_peripheral_50000000_S_sai_107000_P_pinmuxes_IDX_0_VAL_pin'?
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 13347 | #define DT_N_INST_1_nxp_mcux_i2s DT_N_S_soc_S_peripheral_50000000_S_sai_107000
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 72 | #define __DEBRACKET(...) __VA_ARGS__
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                          ^~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:4956:9: note: in expansion of macro 'COND_CODE_1'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 4956 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:1565:9: note: in expansion of macro 'DT_CAT7'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1565 |         DT_CAT7(node_id, _P_, pha, _IDX_, idx, _VAL_, cell)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |         ^~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1157:36: note: in expansion of macro 'DT_PHA_BY_IDX'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1157 |                 .mclk_pin_offset = DT_PHA_BY_IDX(DT_DRV_INST(i2s_id), pinmuxes, 0, offset),        \
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                    ^~~~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_1_nxp_mcux_i2s'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                    ^
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                          ^~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:349:31: note: in expansion of macro 'UTIL_CAT'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 349 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                               ^~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:3879:27: note: in expansion of macro 'DT_INST'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 3879 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                           ^~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1157:50: note: in expansion of macro 'DT_DRV_INST'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1157 |                 .mclk_pin_offset = DT_PHA_BY_IDX(DT_DRV_INST(i2s_id), pinmuxes, 0, offset),        \
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                                  ^~~~~~~~~~~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_mcxn947_mcxn947_cpu0/zephyr/tests/drivers/i2s/i2s_speed/drivers.i2s.speed/zephyr/include/generated/zephyr/devicetree_generated.h:27497:53: note: in expansion of macro 'I2S_MCUX_INIT'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 27497 | #define DT_FOREACH_OKAY_INST_nxp_mcux_i2s(fn) fn(0) fn(1)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                                     ^~
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_mcux_i2s'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - |                                    ^
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/i2s/i2s_mcux_sai.c:1222:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 1222 | DT_INST_FOREACH_STATUS_OKAY(I2S_MCUX_INIT)
2025-01-26 16:03:51,274 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-01-26 16:03:51,275 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - ninja: build stopped: subcommand failed.
2025-01-26 16:03:51,275 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - 
2025-01-26 16:03:51,275 - __main__ -zephyr_ci_build_deploy_test.py : 188 - INFO - INFO    - /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_mcxn947_mcxn947_cpu0/zephyr/tests/drivers/i2s/i2s_speed/drivers.i2s.speed/build.log
```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used:  v4.0.0-3992-gf0d0264c0572

--- a/dts/bindings/i2s/nxp,mcux-i2s.yaml
+++ b/dts/bindings/i2s/nxp,mcux-i2s.yaml
@@ -66,5 +66,5 @@ properties:
     description: Clock mux source for SAI root clock
 
 pinmux-cells:
-  - pin
-  - function
+  - offset
+  - mask

